### PR TITLE
Simplified converter error messages and added resource address

### PR DESCRIPTION
**Pull request by melinath**
Goals:

- Reduce technical jargon so that errors are easier to understand
- Add resource address to all error messages so that large plans are easier to debug.
- Don't log anything related to non-google resources
- Reduce level of messages that don't impact normal operations
- Add debug-level warnings if there is a beta resource in the plan

b/225383221

### CODE REVIEW

File: converters/google/convert.go [hunk 1]
```diff
@@ -173,33 +173,43 @@ type Converter struct {
...
 	for _, rc := range changes {
-		// skip unknown resources
+		// Silently skip non-google resources
+		if !strings.HasPrefix(rc.Type, "google_") {
+			continue
+		}
+
+		// Warn about google-beta resources
+		if rc.ProviderName == "registry.terraform.io/hashicorp/google-beta" {
+			c.errorLogger.Debug(fmt.Sprintf("%s: resource uses the google-beta provider and may not be convertable", rc.Address))
```
Comment (roaks3) [inline converters/google/convert.go:183]: Nit: Not sure if there is precedence with "convertable", but I believe it technically should be "convertible"

File: converters/google/convert.go [hunk 2]
```diff
@@ -240,21 +250,21 @@ func (c *Converter) addDelete(rc *tfjson.ResourceChange) error {
...
-					c.errorLogger.Warn(fmt.Sprintf("%s did not return a value for ID field. Skipping asset fetch.", key))
+					c.errorLogger.Debug(fmt.Sprintf("%s: Unable to fetch and merge remote %s asset due to unset or (known after apply) identity fields on the TF resource.", rc.Address, converted.Type))
 					existingConverterAsset = nil
 				} else if errors.Cause(err) == resources.ErrResourceInaccessible {
-					c.errorLogger.Warn(fmt.Sprintf("%s was not able to be fetched due to not existing or insufficient permission. Skipping asset fetch.", key))
+					c.errorLogger.Warn(fmt.Sprintf("%s: Fetching %s for merge failed due to not existing or insufficient permission.", rc.Address, key))
 					existingConverterAsset = nil
 				} else if err != nil {
-					return errors.Wrap(err, "fetching asset")
+					return fmt.Errorf("fetching remote asset %s: %w", key, err)
```
Comment (roaks3) [inline converters/google/convert.go:259]: It seems intentional for a few of the conditional blocks you have, but I'm curious why we are not prefixing these errors with an `rc.Address` like the others

File: converters/google/convert_test.go [hunk 3]
```diff
@@ -26,22 +27,49 @@ import (
...
+type bufferWriteSyncer struct {
+	*bytes.Buffer
+}
+
+func (bws bufferWriteSyncer) Sync() error {
+	return nil
+}
+
+// This returns a logger to allow deterministic testing of the output
+// by omitting the timestamp and calling function.
```
Comment (roaks3) [inline converters/google/convert_test.go:44]: ❤️

**Review by roaks3 — APPROVED** [review]
LGTM, nothing blocking, but I added one nit and one question

File: converters/google/convert_test.go (hunk 3 above)
Comment (melinath) [reply converters/google/convert_test.go:44]: This code is pretty similar to the zap.NewExample code - tbh I'm a little unsure it is actually deterministic 😂

File: converters/google/convert.go (hunk 1 above)
Comment (melinath) [reply converters/google/convert.go:183]: nope, no precedent. just misspelled it. fixed in 414602ab

**Pull request merged**